internal/api/connectors/inventory/inventory.go

```diff
@@ -0,0 +1,175 @@
+package inventory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"playbook-dispatcher/internal/common/constants"
+	"playbook-dispatcher/internal/common/utils"
+	"time"
+
+	"github.com/redhatinsights/platform-go-middlewares/request_id"
+	"github.com/spf13/viper"
+)
+
+const basePath = "/api/inventory/"
+
+type inventoryConnectorImpl struct {
+	client ClientWithResponsesInterface
+}
+
+func getSatelliteFacts(facts *[]FactSet) SatelliteFacts {
+	satelliteFacts := SatelliteFacts{}
+	for _, fact := range *facts {
+		if fact.Namespace == "satellite" {
+			satelliteInstanceID, idExists := fact.Facts["satellite_instance_id"].(string)
```

Comment:
I _think_ this will panic if don't have a "satellite_instance_id" (or "satellite_version") field.

I think this needs to happen as 2 steps:  1) look for the key/value in the map 2) convert the value to a string